clickhouse-http-client/src/main/java/com/clickhouse/client/http/ClickHouseHttpConnection.java

```diff
@@ -417,10 +410,23 @@ protected String getDefaultUserAgent() {
         return config.getClientName();
     }
 
+    protected String getAdditionalFrameworkUserAgent() {
+        List<String> frameworks = List.of("apache.spark");
+        Set<String> inferredFrameworks = new LinkedHashSet<>();
+        for (StackTraceElement ste : Thread.currentThread().getStackTrace()) {
+            for (String framework : frameworks) {
+                if (ste.toString().contains(framework)) {
+                    inferredFrameworks.add(String.format("(%s)", framework));
```

Comment:
Is the idea to collect all framework, can you provide an example?

Reply:
or we should break once we detected one of them

Reply:
IMHO we don't want to limit ourselves to a single framework use case. A nested or co-shared use case might be possible. From a complexity point of view, it is kind of the same, therefore I would prefer us not to limit this.

Reply:
I think, we need to exit loop when framework is detected. 
And may be remember it to not search again.

Reply:
@chernser 
You must search every time. Each request in an application doesn't necessarily mean we use the framework (you can write s Spark application, work with Spark JDBC, and later on open a regular JDBC connection and query CH).

In addition, could you elaborate on why do we need to exit the loop if a framework was detected? what if a user might use a combination of frameworks that we want to detect? (because as I said, from a complexity point of view there isn't much difference (the list of FRAMEWORKS_TO_INFER is going to be limited), so what is the rationality behind limiting this ability?)